homeassistant/components/climate/__init__.py

```diff
@@ -288,6 +300,68 @@ class ClimateEntity(Entity, cached_properties=CACHED_PROPERTIES_WITH_ATTR_):
     _attr_target_temperature: float | None = None
     _attr_temperature_unit: str
 
+    __mod_supported_features: ClimateEntityFeature = ClimateEntityFeature(0)
+
+    def __getattribute__(self, __name: str) -> Any:
+        """Get attribute.
+
+        Modify return of `supported_features` to
+        include `_mod_supported_features` if attribute is set.
+        """
+        if __name != "supported_features":
+            return super().__getattribute__(__name)
+        _supported_features = super().__getattribute__(__name)
+
+        # Implements check for deprecated use of ClimateEntityFeature as int
+        if type(_supported_features) is int:  # noqa: E721
+            new_features = ClimateEntityFeature(_supported_features)
+            self._report_deprecated_supported_features_values(new_features)
+        return super().__getattribute__(__name) | super().__getattribute__(
+            "_ClimateEntity__mod_supported_features"
+        )
+
+    @callback
+    def add_to_platform_start(
+        self,
+        hass: HomeAssistant,
+        platform: EntityPlatform,
+        parallel_updates: asyncio.Semaphore | None,
+    ) -> None:
+        """Start adding an entity to a platform."""
+        super().add_to_platform_start(hass, platform, parallel_updates)
+
+        def _report_turn_on_off(feature: str) -> None:
+            """Log warning not implemented turn on/off feature."""
+            report_issue = self._suggest_report_issue()
+            _LOGGER.warning(
+                (
+                    "Entity %s (%s) has not implemented ClimateEntityFeature.%s supported feature"
+                    " but is implementing the %s service call. Please %s"
+                ),
+                self.entity_id,
+                type(self),
+                feature,
+                feature.lower(),
+                report_issue,
+            )
+
+        # Adds ClimateEntityFeature.TURN_OFF/TURN_ON depending on service calls implemented
+        # This should be removed in 2025.1.
+        if type(self).async_turn_off is not ClimateEntity.async_turn_off or hasattr(
+            self, "turn_off"
+        ):
```

Comment:
The PR which adds the toggle method adds turn_off + turn_on to the base class. Should we maybe do it already here, otherwise that PR will have to be careful to not break the checks here.